scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -214,10 +236,16 @@
   [{:keys [existing-state-map name] :as kcc} ^V1Pod new-pod]
   (let [pod-name (api/V1Pod->name new-pod)]
     (locking (calculate-lock pod-name)
-      (let [new-state {:pod new-pod :synthesized-state (api/pod->synthesized-pod-state new-pod)}
+      (let [new-state {:pod new-pod
+                       :synthesized-state (api/pod->synthesized-pod-state new-pod)
+                       :sandbox-file-server-container-state (api/pod->sandbox-file-server-container-state new-pod)}
             old-state (get @existing-state-map pod-name)]
         ; We always store the updated state, but only reprocess it if it is genuinely different.
         (swap! existing-state-map assoc pod-name new-state)
+        (let [new-file-server-state (:sandbox-file-server-container-state new-state)
+              old-file-server-state (:sandbox-file-server-container-state old-state)]
+          (when (and (= new-file-server-state :running) (not= old-file-server-state :running))
+            (record-sandbox-url new-state)))
```

Comment:
Is it possible for us to completely miss the `running` state and thus never record the sandbox url? Is there a way we could avoid that by being more defensive here?

Reply:
We'll revisit later